gnome-initial-setup/pages/live-chooser/gis-live-chooser-page.c

```diff
@@ -0,0 +1,405 @@
+  language_name = gnome_get_language_from_locale (self->language, self->language);
+
+  setlocale (LC_MESSAGES, self->language);
+  setlocale (LC_TIME, self->language);
+
+  gtk_label_set_label (OBJ (GtkLabel*, "selected_language_label"), language_name);
+  g_free (language_name);
+
+  /* gis spawns processes that also need to be localised */
+  g_setenv ("LC_MESSAGES", self->language, TRUE);
+  g_setenv ("LC_TIME", self->language, TRUE);
+
+  if (gis_driver_get_mode (driver) == GIS_DRIVER_MODE_NEW_USER)
+    {
+      if (g_permission_get_allowed (self->permission))
+        set_localed_locale (self);
+      else if (g_permission_get_can_acquire (self->permission))
+        g_permission_acquire_async (self->permission, NULL, change_locale_permission_acquired, self);
+    }
+
+  user = act_user_manager_get_user (self->act_client, g_get_user_name ());
+
+  if (act_user_is_loaded (user))
+    act_user_set_language (user, self->language);
+  else
+    g_signal_connect (user, "notify::is-loaded", G_CALLBACK (user_loaded), g_strdup (self->language));
+
+  gis_driver_set_user_language (driver, self->language);
+}
+
+static void
+language_changed (CcLanguageChooser  *chooser,
+                  GParamSpec         *pspec,
+                  GisLiveChooserPage *self)
+{
+  set_language (self);
+  gis_driver_locale_changed (GIS_PAGE (self)->driver);
+}
+
+
+static void
+load_css_overrides (GisLiveChooserPage *self)
+{
+  GtkCssProvider *provider;
+  GError *error;
+  GFile *file;
+
+  error = NULL;
+  provider = gtk_css_provider_new ();
+  file = g_file_new_for_uri ("resource:///org/gnome/initial-setup/live-chooser.css");
+  gtk_css_provider_load_from_file (provider, file, &error);
+  g_object_unref (file);
+
+  if (error != NULL)
+    {
+      g_warning ("Unable to load CSS overrides for the live-chooser page: %s", error->message);
+      g_clear_error (&error);
+    }
+  else
+    {
+      gtk_style_context_add_provider_for_screen (gtk_widget_get_screen (GTK_WIDGET (self)),
+                                                 GTK_STYLE_PROVIDER (provider),
+                                                 GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+    }
+
+  g_object_unref (provider);
+}
+
+static void
+update_assistant (GisLiveChooserPage *self)
+{
+  GisAssistant *assistant = gis_driver_get_assistant (GIS_PAGE (self)->driver);
+
+  gis_assistant_next_page (assistant);
+}
+
+static void
+try_button_clicked (GisLiveChooserPage *self)
+{
+  create_live_user (self);
+  update_assistant (self);
+}
+
+static void
+install_button_clicked (GisLiveChooserPage *self)
+{
+  GError *error = NULL;
+
+  g_spawn_command_line_async ("/usr/lib/eos-installer/gnome-image-installer", &error);
+
+  if (error)
+    {
+      g_critical ("Error running the installer. This should not happen.");
+      g_clear_error (&error);
+      return;
+    }
+
+  gis_driver_hide_window (GIS_PAGE (self)->driver);
+  g_application_quit (G_APPLICATION (GIS_PAGE (self)->driver));
```

Comment:
I think we should wait for `g_spawn_command_line_async` to return before ending the process.